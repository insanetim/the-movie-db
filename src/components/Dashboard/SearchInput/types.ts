export interface SearchInputProps {
  query: string
}

export interface SearchInputHookProps extends SearchInputProps {}

export interface SearchInputHook {
  currentValue: string
  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void
  handleSearch: (value: string) => void
}
