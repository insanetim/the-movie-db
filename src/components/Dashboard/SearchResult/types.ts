import {
  ErrorMsg,
  HandlePagination,
  Query,
} from 'src/interfaces/global.interface'
import { IMoviesList } from 'src/interfaces/movie.interface'

export type SearchResultHookProps = SearchResultProps

export type SearchResultHookReturn = {
  error: ErrorMsg
  handlePagination: HandlePagination
  isLoading: boolean
  movies: IMoviesList | undefined
}

export type SearchResultProps = {
  query: Query
}
