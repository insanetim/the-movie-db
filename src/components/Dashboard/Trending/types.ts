import type { IMoviesList } from 'src/interfaces/movie.interface'

export interface TrendingHook {
  error: null | string
  handlePagination: (page: number) => void
  loading: boolean
  movies: IMoviesList | null
}
