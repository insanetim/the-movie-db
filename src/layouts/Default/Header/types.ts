import { IAccount } from 'src/interfaces/account.interface'

export type HeaderHookReturn = {
  account: IAccount | null
  handleLogOut: () => Promise<void>
}
