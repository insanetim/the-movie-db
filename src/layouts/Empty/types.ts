import { ReactNode } from 'react'

export type EmptyLayoutProps = {
  children?: ReactNode
}
