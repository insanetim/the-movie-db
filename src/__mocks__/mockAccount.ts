import { IAccount } from 'src/interfaces/account.interface'

const mockAccount: IAccount = {
  avatar: {
    gravatar: {
      hash: '/image'
    },
    tmdb: {
      avatar_path: null
    }
  },
  id: 1234,
  include_adult: true,
  iso_639_1: 'en',
  iso_3166_1: 'en',
  name: '',
  username: 'test/username'
}

export default mockAccount
