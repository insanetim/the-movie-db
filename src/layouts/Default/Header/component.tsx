import { CaretDownOutlined, UserOutlined } from '@ant-design/icons'
import { Avatar, Col, Dropdown, Layout, MenuProps, Row, Typography } from 'antd'
import { Link } from 'react-router-dom'
import ThemeSwitch from 'src/components/ThemeSwitch'
import { APP_NAME } from 'src/constants/app'
import { getAvatarUrl } from 'src/utils/helpers/getAvatarUrl'

import useContainer from './hook'

const Header: React.FC = () => {
  const { account, handleLogin, handleLogout, sessionId } = useContainer()

  const items: MenuProps['items'] = [
    { key: 'item-1', label: <Link to='/'>Dashboard</Link> },
    { key: 'item-2', type: 'divider' },
    { key: 'item-3', label: <Link to='/lists'>My Lists</Link> },
    { key: 'item-4', label: <Link to='/favorite'>Favorite</Link> },
    { key: 'item-5', label: <Link to='/watchlist'>Watchlist</Link> },
    { key: 'item-6', type: 'divider' },
    { key: 'item-7', label: 'Sign out', onClick: handleLogout },
  ]

  let avatar: JSX.Element
  if (account) {
    avatar = (
      <Avatar
        alt='User avatar'
        src={getAvatarUrl(account)}
      />
    )
  } else {
    avatar = <Avatar icon={<UserOutlined />} />
  }

  const username = account?.username ? account.username : 'Username'

  return (
    <Layout.Header>
      <Row justify='space-between'>
        <Col>
          <Link
            className='logo text-white'
            to='/'
          >
            {APP_NAME}
          </Link>
        </Col>
        <Col>
          <Row gutter={24}>
            <Col>
              <ThemeSwitch />
            </Col>
            <Col>
              {sessionId ? (
                <Dropdown menu={{ items }}>
                  <Typography.Text className='text-white'>
                    {avatar}
                    {` `}
                    <span className='hide-sm-down'>{username}</span>
                    {` `}
                    <CaretDownOutlined />
                  </Typography.Text>
                </Dropdown>
              ) : (
                <Typography.Link
                  className='text-white'
                  onClick={handleLogin}
                >
                  Sign in
                </Typography.Link>
              )}
            </Col>
          </Row>
        </Col>
      </Row>
    </Layout.Header>
  )
}

export default Header
