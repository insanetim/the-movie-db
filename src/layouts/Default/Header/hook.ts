import { useEffect } from 'react'
import { useSelector } from 'react-redux'
import { useLocation, useNavigate } from 'react-router-dom'
import { useAppDispatch } from 'src/hooks/useRedux'
import useScrollToTop from 'src/hooks/useScrollToTop'
import { fetchAccount, logOut } from 'src/store/auth/actions'
import { accountSelector } from 'src/store/auth/selectors'

import { HeaderHookReturn } from './types'

const useContainer = (): HeaderHookReturn => {
  const dispatch = useAppDispatch()
  const account = useSelector(accountSelector)
  const location = useLocation()
  const navigate = useNavigate()
  useScrollToTop()

  const handleLogOut = async () => {
    dispatch(logOut())
    navigate('/login', {
      replace: true,
      state: { from: location },
    })
  }

  useEffect(() => {
    if (!account) {
      dispatch(fetchAccount())
    }
  }, [account, dispatch])

  return { account, handleLogOut }
}

export default useContainer
