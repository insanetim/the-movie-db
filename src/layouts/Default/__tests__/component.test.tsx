import mockAccount from 'src/__mocks__/mockAccount'
import renderWithWrapper from 'src/utils/testHelpers/renderWithWrapper'

import { HeaderHookReturn } from '../Header/types'
import DefaultLayout from '../component'

const mockedHeader: HeaderHookReturn = {
  account: mockAccount,
  handleLogIn: jest.fn(),
  handleLogOut: jest.fn(),
  isAuthenticated: true,
}
jest.mock('../Header/hook', () => jest.fn(() => mockedHeader))

describe('DefaultLayout component', () => {
  it('should match snapshot', () => {
    const { asFragment } = renderWithWrapper(<DefaultLayout />)

    expect(asFragment()).toMatchSnapshot()
  })
})
