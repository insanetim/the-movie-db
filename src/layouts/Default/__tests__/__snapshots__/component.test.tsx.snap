// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-layout css-dev-only-do-not-override-1c0na6j"
  >
    <header
      class="ant-layout-header css-dev-only-do-not-override-1c0na6j"
    >
      <div
        class="ant-row ant-row-space-between css-dev-only-do-not-override-1c0na6j"
      >
        <div
          class="ant-col css-dev-only-do-not-override-1c0na6j"
        >
          <a
            class="logo text-white"
            href="/"
          >
            The Movie DB
          </a>
        </div>
        <div
          class="ant-col css-dev-only-do-not-override-1c0na6j"
        >
          <span
            class="ant-typography ant-dropdown-trigger css-dev-only-do-not-override-1c0na6j"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-1c0na6j"
            >
              <img
                alt="User avatar"
                src="https://www.gravatar.com/avatar/image"
              />
            </span>
             
            <span
              class="hide-sm-down"
            >
              test/username
            </span>
             
            <span
              aria-label="caret-down"
              class="anticon anticon-caret-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="caret-down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
    </header>
    <main
      class="ant-layout-content css-dev-only-do-not-override-1c0na6j"
    />
    <footer
      class="ant-layout-footer css-dev-only-do-not-override-1c0na6j"
    >
      <span
        style="opacity: 0.8;"
      >
        Made with ‚ù§ by 
        <a
          class="ant-typography css-dev-only-do-not-override-1c0na6j"
          href="https://github.com/insanetim/the-movie-db"
          rel="noreferrer"
          target="_blank"
        >
          insanetim
        </a>
      </span>
    </footer>
  </div>
</DocumentFragment>
`;
